### DISCUSSION TITLE
input security in Python programming?

### DISCUSSION THREAD

**Discussion by Leilamodaresi** [community/community#55456] [Code Security]
### Select Topic Area

Question

### Body

What are the solutions to increase input security in Python programming and prevent injection attacks?

**Comment by ericpd** [comment]
Considering that injection attacks, such as SQL Injection and Command Injection, are among the most common types of attacks in programming, to prevent them in Python programming, you can use the following methods:

1- Validation of input data: User input data must be validated before use. For example, you can use the Python re library to validate user emails and check email patterns.

2- Filtering inputs: Your program should not use user-generated data directly in the program and should be filtered before use. For example, to filter data uploaded from a file, you can use the werkzeug library in Python.

3- Use of SQL Parameters: To prevent SQL injection attacks, you can use SQL parameters in your SQL commands. Using this method, input data is not directly added to the SQL command and must first be defined parametrically.

4- Running the program with lower privileges: To prevent command injection, your program must be executed with lower privileges. For example, using the subprocess library with lower privileges instead of os.system with higher privileges can help.

5- Use of secure libraries: In Python, there are many libraries designed for the security of input data. For example, the pyinputplus library is optimized for managing user input in Python.

Using the above methods increases input security in Python programming and prevents injection attacks

**Comment by queenofcorgis** [comment]
@Leilamodaresi @ericpd GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answering and locking the post. Please note any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
